docs/development/sync-protocol.md

```diff
@@ -0,0 +1,431 @@
+				"ou": "Apple Certification Authority",
+				"sha256": "b0b1730ecbc7ff4505142c49f1295e6eda6bcaed7e2c68c5be91b5a11001f024"
+			}
+		],
+		"file_bundle_name": "santasyncservice",
+		"executing_user": "root",
+		"ppid": 1,
+		"file_bundle_path": "/Applications/Santa.app",
+		"file_name": "santasyncservice",
+		"execution_time": 1657764366.475035,
+		"file_sha256": "8621d92262aef379d3cfe9e099f287be5b996a281995b5cc64932f7d62f3dc85",
+		"decision": "ALLOW_BINARY",
+		"file_bundle_id": "com.google.santa.syncservice",
+		"file_bundle_version_string": "9999.1.1",
+		"pid": 2595,
+		"current_sessions": [
+			"markowsky@console",
+			"markowsky@ttys000",
+			"markowsky@ttys001",
+			"markowsky@ttys003"
+		],
+		"team_id": "XXXXXXXXXX"
+	}]
+}
+```
+
+#### `eventupload` Response
+
+The server should reply with an HTTP 200 if the request was successfully received and processed.  
+
+
+| Key | Required | Type | Meaning | Example Value |
+|---|---|---|---|---|
+| event_upload_bundle_binaries | NO | list of strings | An array of bundle hashes that the sync server needs to be uploaded |  |
+
+##### `eventupload` Response Example Payload 
+
+
+```json
+{
+   "event_upload_bundle_binaries": ["8621d92262aef379d3cfe9e099f287be5b996a281995b5cc64932f7d62f3dc85"]
+}
+```
+
+### Rule Download
+
+After Events have been uploaded to the Sync server, the Client then initiates the `rulesdownload` request. 
+
+Like the previous stages this is a simple HTTP request response cycle like so:
+
+```mermaid
+sequenceDiagram
+   client ->> server: POST /ruledownload/<machine_id>
+   server -->> client: ruledownload response
+```
+
+This stage may be performed many times depending on the rule batch size set during the `preflight` stage. 
+
+In the event that the client requested a clean sync or the server responded with a `clean_sync` in the preflight stage then it is expected that the client will throw out its rules and the sync server should supply the client with all of the rules configured for it. 
+
+If a clean sync was not requested by either the client or the sync service, then the sync service should only send new rules seen since the last time the client synced.
+
+#### `ruledownload` Request
+
+Rules are retrieved from the sync server by having the client (Santa) issues an
+HTTP POST request to the url `/ruledownload/<machine_id>`
+
+| Key | Required | Type | Meaning |
+|---|---|---|---|
+| cursor | NO | string | a field used by the Sync Server to indicate where the next batch of rules should start |
+
+
+##### `ruledownload` Request Example Payload 
+
+On the first request the payload is an empty dictionary
+
+```json
+{}
+```
+
+The server will then send a response with the number of rules specified in the preflight response `rule_batch` the response will also contain a special field called `cursor`. The value and form of this field is left to the sync server implementor but is used to indicate that the client is ready for the next batch of rules.
```

Comment:
Number of rules is set by the server and not related to preflight, the client will only make subsequent requests if the cursor field is not empty.

Reply:
Yeah, no idea what I was thinking. I've reworked this section to reflect that.